### ISSUE TITLE
Travis CI Build no longer succeeds

### ISSUE THREAD

**Issue by numtel** [nevill/zongji#26]
The [automated build](https://travis-ci.org/nevill/zongji/builds/101519674) for @vlasky 's PR to add the JSON datatype from MySQL 5.7 has failed but not due to his issue. Travis CI no longer has the `libaio1` or `libaio-dev` packages available which are required to run MySQL server from the tarball as it currently configured. At this current point in time, the Travis CI build will fail if tried to run against `master` for this same reason.

I tried switching the tests to run using the Ubuntu Trusty (14.04) build environment, (it was the default, Precise, 12.04) but [the packages are still not available](https://travis-ci.org/nevill/zongji/builds/101945508).

After this roadblock, I found that there is an [official repository for MySQL server DEB packages](http://dev.mysql.com/doc/mysql-apt-repo-quick-guide/en/#repo-qg-apt-repo-manual-setup). [In progress commit](https://github.com/nevill/zongji/commit/0b6497f3931eb5e8b91cf8dd0abacfed61be9c5a) and [currently failing test results](https://travis-ci.org/nevill/zongji/builds/101966277). It gets to the point of running `npm test` but it's not clear why it fails. The next step will be to enable more logging on the test suite to see where it's failing.

The trouble with this approach is that these official repos only contain packages for MySQL 5.6 and 5.7. We will lose the test runner for MySQL 5.1 and 5.5 unless there's some way to get the `libaio1` and `libaio-dev` packages installed.

I've got to sign off right now but I will keep trying to get this fixed as I have the time.

**Comment by nevill** [comment]
Oh, that's bad news. Probably Docker comes to rescue? I know there are Docker images with different mysql versions. And I also noticed that travis-ci supports with Docker. Checking with [Using Docker In Builds](https://docs.travis-ci.com/user/docker).

**Comment by numtel** [comment]
Yeah, I agree. I'm going to try adding a repo with the packages manually to see if that can fix the current method of extracting each MySQL tarball.

http://packages.ubuntu.com/trusty/amd64/libaio1/download

If that doesn't work, I'll check out the docker path.

**Comment by numtel** [comment]
Good news, adding the extra repo for the packages fixes the current test runner:

https://travis-ci.org/nevill/zongji/builds/102162460

It would be much simpler to use docker images though. I've started on this but it's not passing with 5.6 yet.

https://travis-ci.org/nevill/zongji/builds/102170074

The commit: https://github.com/nevill/zongji/commit/ac28a7960954188b20c071ffccc264a7fa65542d

I'll see if I can get the docker images working. There is 5.5, 5.6, and 5.7 on the official docker repo and 5.1 is available from this user:

https://hub.docker.com/r/mtirsel/mysql-5.1/

I've got the MySQL 5.7 tarball downloaded now but haven't quite got the command sequence figured out yet to get it to run as the current user like the current test runner script. It's taking 30+ minutes to get the Travis results right now, I'll get on again a bit later and hopefully things will be quicker on their end. Using docker would make the Travis build much more maintainable.

**Comment by numtel** [comment]
:smiley_cat: Woot! I've got the Travis builds working using Docker images for the 4 MySQL server versions (5.1, 5.5, 5.6, and 5.7) https://github.com/nevill/zongji/commit/58b12bdb2351a04670d4e90a5d5ba40dafdd0951

https://travis-ci.org/nevill/zongji/builds/102473020

This is a much easier to reason about automated build system now. I've also got the suite running in Node.js 4.1 as well as the original 0.10.

Definitely thanks @nevill for the Docker suggestion!
